README.md

```diff
@@ -60,3 +60,86 @@ Releases are automatically pushed on commits merged to master that fulfill the f
 The release process is building all runtimes and then puts them into a release in this github repository.
 
 The format of [`CHANGELOG.md`](CHANGELOG.md) is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+
+# Weight Generation
+
+To generate weights for a runtime
+
+1. Build `chain-spec-generator` with `--profile production --features runtime-benchmarks`
+2. Use it to build a chain spec for your runtime, e.g. `./target/production/chain-spec-generator --raw polkadot-local > polkadot-chain-spec.json`
+3. Create `file_header.txt`
+
+```text
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+```
+
+4. `rsync` chain spec/s and the file header to a benchmark machine
+
+5. Build `polkadot` binary from the latest release of `polkadot-sdk` with `--profile production --features runtime-benchmarks --bin polkadot` on the benchmark machine
+
+6. Create output directories for the weights on the benchmark machine
+
+7. Run on the benchmark machine:
+
+```bash
+#!/bin/bash
+
+# Default value is 'polkadot', but you can override it by passing a different value as an argument
+CHAIN=${1:-polkadot}
+
+pallets=($(
+  ./target/production/polkadot benchmark pallet --list \
+    --chain=./$CHAIN-chain-spec.json |
+    tail -n+2 |
+    cut -d',' -f1 |
+    sort |
+    uniq
+);
+
+for pallet in "${pallets[@]}"; do
+  echo "Running benchmarks for $pallet"
+  ./target/production/polkadot benchmark pallet \
+    --chain=./$CHAIN-chain-spec.json \
+    --steps=50 \
+    --repeat=20 \
+    --pallet=$pallet \
+    --extrinsic=* \
+    --wasm-execution=compiled \
+    --heap-pages=4096 \
+    --output=./$CHAIN-weights \
```

Comment:
```suggestion
    --output=$output_file \
```

usually, there is hack for xcm benchmarks, which are placed under `*/weights/xcm`:
```
# a little hack for pallet_xcm_benchmarks - we want to force custom implementation for XcmWeightInfo
    if [[ "$pallet" == "pallet_xcm_benchmarks::generic" ]] || [[ "$pallet" == "pallet_xcm_benchmarks::fungible" ]]; then
      output_file="xcm/${pallet//::/_}.rs"
      extra_args="--template=$output_path/templates/xcm-bench-template.hbs"
    fi
```